src/Sentry/SentryOptions.cs

```diff
@@ -450,6 +450,27 @@ public double TracesSampleRate
         /// </remarks>
         public Func<TransactionSamplingContext, double?>? TracesSampler { get; set; }
 
+        private int _maxSpans = 1000;
+
+        /// <summary>
+        /// Maximum spans allowed in a transaction.
+        /// Attempting to add more spans to a transaction than this limit dictates
+        /// will result in the excess spans being silently dropped.
+        /// </summary>
+        public int MaxSpans
```

Comment:
Thanks for pointing out this is experimental in Python.
In hindsight we're be stuck with this setting even if we move towards collecting spans only where this would become a confusing configuration option. 

I propose we just hide this for now. @Tyrrrz @marandaneto what do you think? 

```suggestion
        internal int MaxSpans
```

Reply:
on Java we'llll have an annotation for it.

https://github.com/JetBrains/java-annotations/blob/master/common/src/main/java/org/jetbrains/annotations/ApiStatus.java#L34-L54

which is pretty much the same as Python, but via annotation processing, can we do that instead?
I don't see being useful if its there but hidden, while adding such annotation let us know the possibility of removing later if it was a wrong move, wdyt?

Reply:
I'm personally skeptical about adding public members just to annotate them with "don't use it, it's internal". I can see it as a way to workaround language limitations, like where you need to define a type/member as public because the language requires it to be, rather than because it actually needs to be used publicly. For example, C# doesn't allow public types to implement internal interfaces implicitly, which is a case where "public but not really" annotation would be nice.

In this case, I feel like we'd be overthinking/overengineering a solution to a non-existent problem. Are there any users who actually want to set `MaxSpans` today?

In summary: I think we should either make this option public (and agree on the name) or just not add it at all. In case we choose not to add it, we can still keep `SentryOptions` in `TransactionTracer` for the future.

Reply:
I'm OK with getting rid of it